### DISCUSSION TITLE
Map editor's Navmesh and some files in Environment folder

### DISCUSSION THREAD

**Discussion by Ongjming** [SmartlyDressedGames/Unturned-3.x-Community#2863] [General]
There are some bit of files/data needing a navmesh's number and road's number.
I get that the number goes from 0 to whatever, according to the order of how we initially created them.

There is also the current issue of navmesh number arbitrarily changing itself even upon baking and saving if you moved the navmesh/deleted the navmesh and place it elsewhere. (The baked portion appear to the order you last saved it i.e. if you delete the nav 28 that you see now at somewhere, the nav 28 baked portion of the ground navmap will disappear from the "correct" location.)

There are some solutions out there.
Some said to delete every single navmesh via the map editor, save and exit, then reenter to place them all again from fresh.
Some said to delete the whole content of the Environment folder.
Some said to delete the Navigation files.
Some said to delete Flags/Flags_Data files.
Some said to rename the Navigation files numbers.

While on this track of thought, there are some issue with safezone nodes too at times. It would sometimes be corrupted and made the whole map a safezone. The solution found by some would be to delete the Nodes.dat.

Then there is the 3.21.29.0 update recently,
Fixed:
    Loading levels with mismatched bounds/flags_data/flags files.

Is there going to be someway to manually change a navmesh's number in the far future?
Is there a better solution than what was listed above?
What each of the file do exactly? What do they mean and what it means to delete them?
Is there going to be some way of viewing list of placed objects/placed navmeshs/placed nodes in the far future? (Something like hierarchy view from devkit map editor)

If anyone knows better, thanks if more information is presented/shared :)
This was all I could find on a search through various places.

**Comment by SDGNelson** [comment]
Unfortunately yes, using indices to associate gameplay stuff with roads and navmeshes sucks. Until there is a better persistent ID the only option is to re-use existing roads/navmeshes elsewhere rather than deleting them.

Do you know of a way to reproduce this safezone issue you mentioned?

**Reply by Ongjming** [reply]
The last time I encountered it, I was editing the map quite a bit after I placed the safezone nodes (Placing objects/nodes/item/zombies/baking navmeshes/Adding landscapehole volumes with devkit). Only realised the issue towards the end when I was placing another safezone node prior to testing it out in singleplayer mode.

Hehe, hence the search around the web for people having the same issue + what they did.

Hmm, in short I don't really have a concrete way to keep reproducing this deliberately. Now that I am doing up another map editing, I would probably place the safezone nodes last, just in case.

EDIT: With regards to navmesh reusing, I did tried to move the reordered navmeshes about and rebake them from 0 to whatever number I have. It still is kind of messed up, so currently have since removed them all and saved, and then placing them again.
Was wondering if order of placing matters for the zombie to be chasing players instead of stoning, between:
1. Zombie spawn spot placing, then Navmesh placing and baking.
2. Navmesh placing and baking, then zombie spawn spot placing.

**Reply by SDGNelson** [reply]
Zombie spawns use the navmesh they are inside when the game starts, so you do not need to worry about the order in that case. Just make sure you do not have any navmeshes overlapping.

**Reply by Ongjming** [reply]
Thanks so much! :)
